python/sglang/srt/multimodal/processors/base_processor.py

```diff
@@ -9,11 +10,137 @@
 
 import numpy as np
 import torch
+import transformers
 from PIL import Image
 from transformers import BaseImageProcessorFast
 
 from sglang.srt.managers.schedule_batch import Modality, MultimodalDataItem
-from sglang.srt.utils import load_audio, load_image, load_video, logger
+from sglang.srt.utils import (
+    get_bool_env_var,
+    load_audio,
+    load_image,
+    load_video,
+    logger,
+)
+
+CACHED_IMAGE_MAX_NUM = 64
+QWEN2_VL_PATCH_SIZE = 196.0
+QWEN2_PER_TOKEN_PATCH_NUM = 4
+QWEN2_PATCH_SIZE = 14
+
+VISION_START_TOKEN = 151652
+IMG_PAD_TOKEN = 151655
+VISION_END_TOKEN = 151653
+
+
+def fast_image_hash(img: Image.Image, hash_type: str = "int") -> int | str:
+    small_img = img.resize((8, 8), Image.Resampling.BILINEAR).convert("L")
+    pixel_data = small_img.tobytes()
+    hash_obj = hashlib.md5(pixel_data)
+
+    if hash_type == "int":
+        return int.from_bytes(hash_obj.digest(), byteorder="little", signed=False)
+    else:
+        return hash_obj.hexdigest()
+
+
+def image_to_int(img: Image.Image) -> int:
+    img_bytes = img.tobytes()
+    hash_obj = hashlib.md5(img_bytes)
+    hash_bytes = hash_obj.digest()
+    return int.from_bytes(hash_bytes, byteorder="big")
+
+
+class FIFOHashTable:
```

Comment:
Similar comments as https://github.com/sgl-project/sglang/pull/8742/files#r2284122880

If we should use cache more than once, we can consider `@lru_cache` i think